interceptor.go

```diff
@@ -90,7 +90,9 @@ func (i *interceptor) WrapUnary(next connect.UnaryFunc) connect.UnaryFunc {
 			trace.WithSpanKind(spanKind),
 		)
 		defer span.End()
-		i.config.propagator.Inject(ctx, carrier)
+		if isClient {
+			i.config.propagator.Inject(ctx, carrier)
+		}
```

Comment:
In handlers it doesn't make any sense to inject headers because any sub process would just use `ctx`